### PR TITLE
Added a tutorial

### PR DESCRIPTION
**Pull request by stevehenderson**
I added a step by step tutorial for this excellent repo.

Thank you @mengchaoheng for all the tips in https://github.com/Jaeyoung-Lim/px4-offboard/issues/3

### CODE REVIEW

**Comment by jaypatrikar** [comment]
Hello! Just curious if this has been tried with multiple agents? Thanks

**Comment by stevehenderson** [comment]
> Hello! Just curious if this has been tried with multiple agents? Thanks

Hi @jaypatrikar -- no I haven't tried it yet

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
Thank you for the thorough tutorial

I think overall it looks good, but I am a little uncomfortable of adding binary images to the repository. Would you be able to remove them from the commit, and reduce them / add them as text to the tutorial file?

**Comment by mengchaoheng** [comment]
@Jaeyoung-Lim  binary images is good, why you don't like it?

**Comment by Jaeyoung-Lim** [comment]
@mengchaoheng It is not good practice to have large binaries in trees since it increases the size of the repository.

Also having screenshots of a terminal as a image does not provide more information compared to text

**Comment by mengchaoheng** [comment]
All right. Can you fix it? @stevehenderson

**Comment by stevehenderson** [comment]
> All right. Can you fix it? @stevehenderson

Yes, no problem!

**Comment by stevehenderson** [comment]
> All right. Can you fix it? @stevehenderson

Yes, no problem!

**Comment by Jaeyoung-Lim** [comment]
@stevehenderson Any updates?

**Comment by stevehenderson** [comment]
Hi @Jaeyoung-Lim I will send a revised PR later today

File: doc/ROS2_PX4_Offboard_Tutorial.md [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+
+### Install PX4 ROS Com and PX4 msg
+
+The `px4-offboard` example requires the `px4_ros_com` bridge and `px4_msgs` definitions:
+
+```
+cd ~
+mkdir px4_ros_com_ws
+cd px4_ros_com_ws
+git clone https://github.com/PX4/px4_ros_com.git
```
Comment (Jaeyoung-Lim) [inline doc/ROS2_PX4_Offboard_Tutorial.md:36]: We don't need `px4_ros_com` for this package to work

File: doc/ROS2_PX4_Offboard_Tutorial.md [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+```
+
+### Install PX4 ROS Com and PX4 msg
+
+The `px4-offboard` example requires the `px4_ros_com` bridge and `px4_msgs` definitions:
+
+```
+cd ~
+mkdir px4_ros_com_ws
+cd px4_ros_com_ws
```
Comment (Jaeyoung-Lim) [inline doc/ROS2_PX4_Offboard_Tutorial.md:35]:
```suggestion
cd <colcon_workspace_path>
```

File: doc/ROS2_PX4_Offboard_Tutorial.md [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+```
+Starting >>> px4_msgs
+Finished <<< px4_msgs [6.42s]                     
+Starting >>> px4_ros_com
+Finished <<< px4_ros_com [0.13s]                  
+
+Summary: 2 packages finished [6.71s]
+```
+
+You may see some warnings interspered with the output.  As long as there are no __*errors*__ you should be OK..
```
Comment (Jaeyoung-Lim) [inline doc/ROS2_PX4_Offboard_Tutorial.md:57]:
```suggestion
```

File: doc/ROS2_PX4_Offboard_Tutorial.md [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+source install/local_setup.sh
+ros2 run micro_ros_agent micro_ros_agent udp4 --port 8888
+```
+
+It should respond with:
+
+```
+[1670101284.566587] info     | UDPv4AgentLinux.cpp | init                     | running...             | port: 8888
+[1670101284.566822] info     | Root.cpp           | set_verbose_level        | logger setup           | verbose_level: 4
+```
```
Comment (Jaeyoung-Lim) [inline doc/ROS2_PX4_Offboard_Tutorial.md:179]: Could we just link the instruction to an external link? building a micro-ros-agent is not really related to this package

File: doc/ROS2_PX4_Offboard_Tutorial.md [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+In each terminal window, we will first source any workspace settings required for a particular component and also set the `ROS_DOMAIN_ID` and `PYTHONOPTIMIZE`  in each window
+
+### Start the micro-ros-agent
+
+In the terminal you designated as the `micro-ros-agent` terminal:
+
+```
+cd ~/microros_ws
+export ROS_DOMAIN_ID=0
+export PYTHONOPTIMIZE=1
```
Comment (Jaeyoung-Lim) [inline doc/ROS2_PX4_Offboard_Tutorial.md:201]: Is this necessary?

**Review by Jaeyoung-Lim — CHANGES_REQUESTED** [review]
Thanks for the changes!

I left a few comments that would be nice to address

File: doc/ROS2_PX4_Offboard_Tutorial.md [hunk 6]
```diff
@@ -0,0 +1,415 @@
+
+## Overview
+
+This tutorial explains how to interface ROS2 with PX4 (SITL) using DDS.
+
+### Prerequisites
+
+   * [ROS2 Installed](https://docs.px4.io/main/en/ros/ros2_comm.html#install-ros-2), and setup for your operating system (e.g. [Linux Ubuntu](https://docs.px4.io/main/en/dev_setup/dev_env_linux_ubuntu.html)) with Gazebo
+   * [FastDDS Installed](https://docs.px4.io/main/en/dev_setup/fast-dds-installation.html)
```
[the remark this answers is not in the document]
Comment (stevehenderson) [reply doc/ROS2_PX4_Offboard_Tutorial.md]: Updated.  It doesnt seem fastdds is in the current px4 branch, so I linked to the last supported version.

File: doc/ROS2_PX4_Offboard_Tutorial.md (hunk 1 above)
Comment (stevehenderson) [reply doc/ROS2_PX4_Offboard_Tutorial.md]: Removed

File: doc/ROS2_PX4_Offboard_Tutorial.md (hunk 4 above)
Comment (stevehenderson) [reply doc/ROS2_PX4_Offboard_Tutorial.md]: Done

File: doc/ROS2_PX4_Offboard_Tutorial.md (hunk 5 above)
Comment (stevehenderson) [reply doc/ROS2_PX4_Offboard_Tutorial.md:81]: Yes.  As a newbie, I struggled for hours before I realized that my `ROS_DOMAIN_ID` had to be consistent across the 3 processes.  PYTHONOPTIMIZE is required in my environment for GAZEBO to work.

Comment (Jaeyoung-Lim) [reply doc/ROS2_PX4_Offboard_Tutorial.md:81]: Do you know why PYTHONOPTIMIZE is required? what environment are you on?

Comment (stevehenderson) [reply doc/ROS2_PX4_Offboard_Tutorial.md:81]: My environment:

```
   * Ubuntu 22.02
   * ROS2 Humble
   * Python 3.10
```

When I ran through this the first time, months ago, I got this error:

image

After several hours, I found the solution [here](https://github.com/PX4/px4_ros_com/issues/162)

Comment (Jaeyoung-Lim) [reply doc/ROS2_PX4_Offboard_Tutorial.md:81]: Okay, strange

**Review by Jaeyoung-Lim — APPROVED** [review]

**Pull request merged**